pkg/dashboard/apiserver/event/event.go

```diff
@@ -53,6 +61,7 @@ func Register(r *gin.RouterGroup, s *Service) {
 
 	endpoint.GET("", s.list)
 	endpoint.GET("/:id", s.get)
+	endpoint.GET("/workflow", s.cascadeFetchEventsForWorkflow)
```

Comment:
If we separate into two APIs, it's not easy to adapt. For example, our own grafana plugin has to write additional logic, also in the dashboard events page.

Reply:
> A quick question: why do we need to fetch the workflow related events in a new API?


there are different purposes between `get` and `cascadeFetchEventsForWorkflow`:
-  `get` will fetch all events which belong to ONE object in kubernetes. For example, all events for a NetworkChaos, all events for a Schedule.
- `cascadeFetchEventsForWorkflow` will fetch all events which belongs to Worklflow, and all events which belongs to WorkflowNode controlled by former Workflow.

> Can we put the code into /api/events and pass a param to indicate we want to fetch the workflow events? Like ?kind=workflow. (If not specified, the workflow events will be fetched as default)

So I do not think we could combine them together. 

> If we separate into two APIs, it's not easy to adapt. For example, our own grafana plugin has to write additional logic, also in the dashboard events page.

Actually, `cascadeFetchEventsForWorkflow` is the API only for displaying events of Workflow. I have no idea about using this API in another scenario.
On the other hand, showing events of Workflow and WorkflowNode on Grafana might be meaningless, because it does not do any "chaos injection" actually.

Reply:
Know about it. I originally thought that workflow-related events would have workflow itself as the lowest dimension, but now I see what you mean.